opm/parser/eclipse/Parser/createDefaultKeywordList.cpp

```diff
@@ -34,6 +42,33 @@ int main(int argc , char ** argv) {
         bool verboseLoader = false;
         bool verboseGenerator = true;
 
+        try {
+            /* sometimes the local env's locales are broken on POSIX. Boost <=
+             * 1.56 uses the std::locale("") constructor which respects user
+             * preferred locales, and might crash. If this is the case, and
+             * we're on a non-windows system (assuming POSIX), in case of an
+             * exception, set the environment to "C" and keep going.
+             *
+             * Can be removed once boost < 1.57 is no longer supported
+             */
+            std::locale( "" );
+        } catch( const std::runtime_error& ) {
+            #if !defined(WIN32)
+            setenv( "LC_ALL", "C", 1 );
```

Comment:
It doesn't, and in fact that's how every C++ program is started (by implicitly setting locale == classic). However, that isn't the actual problem here. The problem is that boost calls `std::locale("")` which ignores any locale setting (setting locale with `std::locale` only affects what's returned by `std::locale()`, the default constructor, and there's no (standard) way of changing what's yielded by `std::locale("")`, which on linux picks up on the environment. As long as boost keeps calling `std::locale("")` the environment must either be sane *or* we need to override it, and that is inherently unportable.